components/server/src/workspace/context-parser.ts

```diff
@@ -102,6 +102,15 @@ export interface IPrefixContextParser {
 }
 export const IPrefixContextParser = Symbol("IPrefixContextParser");
 
+// See https://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
+// the magic sequence @{, consecutive dots, leading and trailing dot, ref ending in .lock
+// Adapted https://github.com/desktop/desktop/blob/1e3df9608a834dabdabe793b1b538e334f33c8a1/app/src/lib/sanitize-ref-name.ts
+const invalidCharacterRegex = /[\x00-\x20\x7F~^:?*\[\\|""<>]+|@{|\.\.+|^\.|\.$|\.lock$|\/$/g;
+/** Sanitize a proposed reference name by replacing illegal characters. */
+
+function sanitizedRefName(name: string): string {
+    return name.replace(invalidCharacterRegex, "-").replace(/^[-\+]*/g, "");
+}
```

Comment:
Can anyone help me to get this function to a more proper place? I'm not too familiar with the server codebase and would appreciate a tip

Reply:
I think here is fine actually. We can move it, should we need to reuse